### PR TITLE
fix duplicate name on fedcircle

### PR DESCRIPTION
**Pull request by ArtificialOwl**
fix issues when sharing file on federated circles:

- should fully avoid duplicate name on remotely shared items
- compat with `Files_Sharing/External/Manager`

### CODE REVIEW

File: lib/Migration/Version0031Date20241105133904.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+namespace OCA\Circles\Migration;
+
+use Closure;
+use Doctrine\DBAL\Schema\SchemaException;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+use Psr\Log\LoggerInterface;
+
+class Version0031Date20241105133904 extends SimpleMigrationStep {
```
Comment (come-nc) [inline lib/Migration/Version0031Date20241105133904.php:21]: attribute missing

File: lib/Migration/Version0031Date20241105133904.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	) {
+	}
+
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		try {
+			$table = $schema->getTable('circles_mountpoint');
+			if (!$table->hasIndex('dname')) {
```
Comment (ArtificialOwl) [inline lib/Migration/Version0031Date20241105133904.php:33]: name should be 'mp_sid_hash'

**Review by mejo- — APPROVED** [review]

**Pull request merged**